src/parseOption.ts

```diff
@@ -122,34 +121,38 @@ export function getValue(itemData: Map<any>, field: string) {
  * It will not try to parse value if it is an object. If value is an object, please
  * use prepareCustomFieldsData() instead.
  */
-export function prepareAllFieldsData(itemData: Map<any>, options: Option) {
+export function prepareAllFieldsData(itemData: object, options: Option) {
   const tooltipData: TooltipData[] = [];
 
   // here, fieldOptions still provides format
-  const fieldOptions = d3map(options.fields, function (d) { return d.field; });
+  const fieldOptions = {};
+  if (options && options.fields) {
+    for (let optionField of options.fields) {
+      fieldOptions[optionField.field] = optionField;
+    }
+  }
 
-  itemData.each(function (value: string, field: string) {
-    // prepare title
+  for (let field of Object.keys(itemData)) {
```

Comment:
I think the performance is almost the same. At least looking at jsperf. And for in with hasownproperty is not shorter. I'm okay with leaving this as is.